tests/helpers/test_config_validation.py

```diff
@@ -1462,6 +1462,12 @@ def test_key_value_schemas_with_default() -> None:
         ({"wait_template": "{{ invalid"}, "invalid template"),
         # The validation error message could be improved to explain that this is not
         # a valid shorthand template
+        (
+            {"condition": 123},
+            "Unexpected value for condition: '123'. Expected and, device, not, "
+            "numeric_state, or, state, template, time, trigger, zone, a list of "
+            "conditions or a valid template",
```

Comment:
This is not a good error message considering condition platforms are supported.
It might be better to simplify the message to:
> Expected a condition, a list of conditions or a valid template

Same comment for `cv.CONDITION_SCHEMA`